### PR TITLE
added more info on types etc to contributing doc

### PR DESCRIPTION
**Pull request by nbokulich**

### CODE REVIEW

File: CONTRIBUTING.md
Comment (gregcaporaso) [inline CONTRIBUTING.md]:
```suggestion
You don't need approval from existing QIIME 2 developers or maintainers to create and distribute your own tools, such as plugins, that build on QIIME 2.
This is the beauty of the plugin-based architecture of QIIME 2: development of tools for introducing novel functionality can be distributed by anyone, removing the lead development team as a bottleneck or gatekeeper between developers of new functionality and users interested in that functionality, and customized QIIME 2 distributions can be created and shared by users.
```

Made a couple of edits here, and modified to have one sentence per line (which makes text documents more management with respect to pull request reviews -- in markdown the first newline character is ignored, two are needed for a paragraph break).

File: CONTRIBUTING.md
Comment (gregcaporaso) [inline CONTRIBUTING.md]:
```suggestion
You can also have your users request help on the QIIME 2 Forum, but if you go this route (as opposed to, for example, directing users to your email, GitHub issue tracker, or another forum for technical support) you should monitor the forum (e.g., via forum notifications) to answer user questions about your plugin. 
Please be aware that the forum moderators will not be able to provide support for your plugins and may reach out to you to reply to help requests since they won't necessarily be experts in your tools (and are already tied up providing tech support to the community).
```

File: CONTRIBUTING.md
Comment (gregcaporaso) [inline CONTRIBUTING.md]:
```suggestion
Avoid defining new types and formats when you can reuse existing ones, to maximize compatibility and interoperability (as well as reducing your own time cost!).
A partial list of common semantic types can be found in the [QIIME 2 documentation](https://docs.qiime2.org/), and a complete list of types and formats available in a local installation of QIIME 2 can be accessed with the `qiime tools list-types` and `qiime tools list-formats` commands.
```

File: CONTRIBUTING.md
Comment (gregcaporaso) [inline CONTRIBUTING.md]:
```suggestion
   - Either of these approaches is totally fine.
     The former is an easier starting point.
```

**Review by gregcaporaso — CHANGES_REQUESTED** [review]
@nbokulich, thanks for the review and edits! I have a few suggestions in here - if you accept those I'll merge this, or just let me know if you have questions/etc.

**Comment by nbokulich** [comment]
all approved and committed. Thanks @gregcaporaso ! Let me know if you want other changes, otherwise I leave it to merge when you are ready.

**Review by gregcaporaso — APPROVED** [review]
Thanks again @nbokulich!

**Pull request merged**
